zambeze/orchestration/message/message_activity_validator.py

```diff
@@ -0,0 +1,48 @@
+import logging
+from .abstract_message_validator import AbstractMessageValidator
+from typing import Optional
+
+
+REQUIRED_ACTIVITY_COMPONENTS = {
```

Comment:
Usually it's easier to maintain the code if these JSON/dict fields become properties in an actual Python class. Then we can have a `.to_dict()` or something method that serializes the object of the class into a python dictionary/JSON. 

See this example from a former project: [link](https://github.com/IBM/multi-data-lineage-capture-py/blob/master/src/provlake/model/task_prov_obj.py#L40)

Reply:
Maybe we should consider using @dataclasses more:
https://docs.python.org/3/library/dataclasses.html

Reply:
Thanks for the suggestion, I appreciate the feedback. I started with what you suggested and both the OPTIONAL and REQUIRED COMPONENTS were part of the validator class. The reason I ended up pulling the REQUIRED and OPTIONAL COMPONENTS out was because I did not want to violate the single responsibility principle and I wanted to keep the code DRY.

Both the activity template function and the activity validator need the OPTIONAL and REQUIRED components, does it make sense, do you still th ink it would be clearer to put them back in the class?

> Maybe we should consider using @dataclasses more:
> https://docs.python.org/3/library/dataclasses.html

Is this just a general comment or do you think I missed some concepts that would be beneficial?

Reply:
This is just a general comment for a future improvement. 

We could be using `dataclasses` for classes that have a defined data schema, such as the messages that go through the message brokers. The classes that are stored in SQLite, such as the "ActivityEntity" could also be a dataclass.

We can easily make optional & required fields in dataclasses too.